plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -182,7 +182,7 @@ def main():
         argument_spec=dict(
             api_host=dict(required=True),
             api_user=dict(required=True),
-            api_password=dict(no_log=True),
+            api_password=dict(no_log=True, required=True, fallback=(env_fallback, ['PROXMOX_PASSWORD'])),
```

Comment:
Does ansible test for `required` before or after trying the `fallback` mechanism? I am not sure on that one. If it tests _after_, then this code works as before.

Reply:
Fallbacks are handled before doing anything else, so also before `required` is checked.